nixos/modules/services/backup/borgbackup.nix

```
@@ -20,8 +20,19 @@ let
     lib.concatStringsSep " "
       (lib.mapAttrsToList (x: y: "--keep-${x}=${toString y}") cfg.prune.keep);
 
+  mkExtraArgs = cfg:
+    # Create BASH arrays of extra args
+    lib.concatLines
+      (lib.mapAttrsToList (name: values: ''
+        ${name}=(${values})
```

After looking at this again, I think this should probably be to support spaces in arguments

```suggestion
        ${name}=(${lib.escapeShellArgs values})
```

---

actually, the options are already coerced with this function